validator/validate.go

```diff
@@ -0,0 +1,103 @@
+package validator
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gorilla/schema"
+)
+
+// ErrFormValidationFailed if form validation has failed, if this is ever returned,
+// corresponding call to GetFieldErrors should be made to check for individual field errors
+var ErrFormValidationFailed = errors.New("form validation failed, check field errors")
+
+// FormValidator represents a form validator which can be used to validate an
+// HTML form
+type FormValidator struct {
+	flds      []field
+	fieldErrs map[string][]error
```

Comment:
this bit needs to change to be goroutine safe

i.e., when validation happens, a new instance of this needs to be created and that gets returned by the validate call, otherwise atm multiple goroutines using the same instance will be overwriting/modifying each others field errors